packages/medusa-js/src/resources/admin/stock-locations.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  AdminGetStockLocationsParams,
+  AdminStockLocationsRes,
+  AdminPostStockLocationsLocationReq,
+  AdminPostStockLocationsReq,
+  AdminStockLocationsListRes,
+} from "@medusajs/medusa"
+import { ResponsePromise } from "../../typings"
+import BaseResource from "../base"
+import qs from "qs"
+
+class AdminStockLocationsResource extends BaseResource {
+  /** retrieve an stock location
+   * @experimental This feature is under development and may change in the future.
+   * To use this feature please install @medusajs/stock-location
+   * @description gets a medusa stock location
+   * @returns a medusa stock location
+   */
+  create(
+    payload: AdminPostStockLocationsReq,
+    customHeaders: Record<string, any> = {}
+  ): ResponsePromise<AdminStockLocationsRes> {
+    const path = `/admin/stock-locations`
+    return this.client.request("GET", path, payload, {}, customHeaders)
```

Comment:
```suggestion
    return this.client.request("POST", path, payload, {}, customHeaders)
```

todo: comments need to be updated